### PR TITLE
Improve SSRF detection in apache http client

### PR DESCRIPTION
**Pull request by manuel-alvarez-alvarez**
# What Does This Do
Improves SSRF accuracy when using apache http client

# Motivation

# Additional Notes

# Contributor Checklist

- [ ] Format the title [according the contribution guidelines](/CONTRIBUTING.md#title-format)
- [ ] Assign the `type:` and (`comp:` or `inst:`) labels in addition to [any usefull labels](/CONTRIBUTING.md#labels)
- [ ] Squash your commits prior merging or merge using GitHub's [Squash and merge](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/about-pull-request-merges#squash-and-merge-your-commits)
- [ ] Don't use `close`, `fix` or any [linking keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) when referencing an issue.  
  Use `solves` instead, and assign the PR [milestone](https://github.com/DataDog/dd-trace-java/milestones) to the issue
- [ ] Update the [public documentation](https://docs.datadoghq.com/tracing/trace_collection/library_config/java/) in case of new configuration flag or behavior

Jira ticket: [PROJ-IDENT]

### CODE REVIEW

File: dd-java-agent/instrumentation/apache-httpclient-4/build.gradle [hunk 1]
```diff
@@ -41,6 +41,7 @@ dependencies {
   iastIntegrationTestRuntimeOnly(project(':dd-java-agent:instrumentation:jetty-9'))
   iastIntegrationTestRuntimeOnly(project(':dd-java-agent:instrumentation:apache-httpcore-4'))
   iastIntegrationTestRuntimeOnly(project(':dd-java-agent:instrumentation:servlet'))
+  iastIntegrationTestRuntimeOnly(project(':dd-java-agent:instrumentation:java-lang'))
```
Comment (jandro996) [inline dd-java-agent/instrumentation/apache-httpclient-4/build.gradle:44]: Is this to keep propagation in testing?

**Review by jandro996 — APPROVED** [review]

**Comment by jandro996** [comment]
We need more test to pass `':dd-java-agent:agent-iast:jacocoTestCoverageVerification'`

`Rule violated for class com.datadog.iast.sink.SsrfModuleImpl.SsrfEvidenceBuilder: branches covered ratio is 0.5, but expected minimum is 0.6`

File: dd-java-agent/instrumentation/apache-httpclient-4/build.gradle (hunk 1 above)
Comment (manuel-alvarez-alvarez) [reply dd-java-agent/instrumentation/apache-httpclient-4/build.gradle:44]: Yep, exactly.

**Comment by manuel-alvarez-alvarez** [comment]
> We need more tests to pass `':dd-java-agent:agent-iast:jacocoTestCoverageVerification'`
> 
> `Rule violated for class com.datadog.iast.sink.SsrfModuleImpl.SsrfEvidenceBuilder: branches covered ratio is 0.5, but expected minimum is 0.6`

Yep, this should be fixed in the last commit

**Pull request merged**